File: Fnd/src/app/services/product.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class ProductService {

  constructor(private _http: HttpClient) { }
  

  //load all the products
  public products()
  {
    return this._http.get(`${baseUrl}/product/`);
  }

  //add new product
  public addProduct(product) {
    return this._http.post(`${baseUrl}/product/`, product);
  }

  //delete product
  public deleteProduct(pid) {
    return this._http.delete(`${baseUrl}/product/${pid}`)
  }

  //get product
  public getProduct(pid) {
    return this._http.get(`${baseUrl}/product/${pid}`);
  }

  //update product
  public updateProduct(product) {
    return this._http.put(`${baseUrl}/product/`, product);
  }

  //get product
  public getProductOfCategory(cid) {
    return this._http.get(`${baseUrl}/product/category/${cid}`)
  }
}

File: Bnd/src/main/java/com/example/demo/service/ProductService.java
package com.example.demo.service;

import java.util.List;
import java.util.Set;

import com.example.demo.dto.ProductDto;
import com.example.demo.model.ecommerce.Category;
import com.example.demo.model.ecommerce.Product;

public interface ProductService {

	public Product addProduct(Product product);

	public Product updateProduct(ProductDto productDto);

	public Set<Product> getProducts();

	public Product getProduct(Long productId);

	public void deleteProduct(Long productId);

	public List<Product> getProductOfCategory(Category category);

}
